sky/backends/backend_utils.py

```diff
@@ -1559,6 +1559,7 @@ def check_owner_identity(cluster_name: str) -> None:
 
     cloud = handle.launched_resources.cloud
     current_user_identity = cloud.get_current_user_identity()
+    supported_user_identities = cloud.get_supported_identities()
```

Comment:
Should we directly make `get_current_user_identity` as a list, instead of adding a new API?

Reply:
I was a little confused by our implementation (and perhaps there's a bug there) which made me add this new method.

Currently, we iterate on the `zip(owner_identity, current_user_identity)` to check identities:

https://github.com/skypilot-org/skypilot/blob/2e545b8db70991bc74d112d5070fa908078f9cdb/sky/backends/backend_utils.py#L1578-L1586

Say the user had launched the cluster with just one identity configured in their kubeconfig, and thus `owner_identity` is `["myid1"]`. 

Say they updated their kubeconfig and now they have 3 identities in `current_user_identity`, the original one being 5th in the list: `["myid3", "myid2", "myid1"]`. 

`zip(owner_identity, current_user_identity)` will now be `["myid1", "myid3"]`, which will fail the check we have.

```
>>> x=["myid1"]
>>> y=["myid3", "myid2", "myid1"]
>>> list(zip(x,y))
[('myid1', 'myid3')]
```

For k8s this logic should be changed to [itertools.product](http://docs.python.org/library/itertools.html#itertools.product) instead of `zip`, but I think that will affect AWS implementation, which has a stricter ordering requirement.

Reply:
The reason we have a list for the `current_user_identity` and `owner_identity` is that it can be fine for a cloud owner with a different cloud account but from the same project to operate a VM, for example, we allows the following to match
```python
owner_identity = ['myaccount1', 'myproject']
current_identity = ['myaccount2', 'myproject']
for i, (owner, 
         current) in enumerate(zip(owner_identity, 
                                   current_user_identity)): 

```

In order to support multiple `current_identity`, we can do something like the following:
```python
owner_identity = ['myaccount1', 'myproject']
current_identity = [['myaccount2', 'myproject']]
for available_identity in current_identities:
  for i, (owner, 
         current) in enumerate(zip(owner_identity, 
                                                      available_identity)): 

```

And for k8s, it can be:
```python
owner_identity = ['mycontext']
current_identity = [['mycontext1'], ['mycontext2']]
for available_identity in current_identities:
  for i, (owner, 
         current) in enumerate(zip(owner_identity, 
                                                      available_identity)): 

```